app/controllers/drinks_controller.rb

```diff
@@ -22,6 +21,38 @@ def show
     @drink = Drink.find(params[:id])
   end
 
+  def edit
+    @drink = Drink.find(params[:id])
+    @liquors = Liquor.all
+    if !signed_in?
+      flash[:error] = "You must be signed in to do that"
+      redirect_to drink_path(@drink)
+    elsif @drink.user != current_user
+      flash[:error] = "You cannot edit someone else's drink"
+      redirect_to drink_path(@drink)
```

Comment:
Seemed like it made sense to have them redirected to the path of the drink they're trying to edit so they can see that it is not their drink.